sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3104,6 +3106,66 @@ class Analyzer(override val catalogManager: CatalogManager)
     }
   }
 
+  object ResolveUserSpecifiedColumns extends Rule[LogicalPlan] {
+    override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperators {
+      case i: InsertIntoStatement if i.table.resolved && i.query.resolved &&
+          i.userSpecifiedCols.nonEmpty =>
+        val resolved = resolveUserSpecifiedColumns(i)
+        val projection = addColumnListOnQuery(i.table.output, resolved, i.query)
+        i.copy(userSpecifiedCols = Nil, query = projection)
+    }
+
+    private def resolveUserSpecifiedColumns(i: InsertIntoStatement): Seq[Attribute] = {
+      val resolved = i.userSpecifiedCols.map {
+        case u: UnresolvedAttribute => withPosition(u) {
```

Comment:
Does other databases support `INSERT INTO t(t.col) ...`? Otherwise using `String` instead of `UnresolvedAttribute` is simpler.

Reply:
yea, it's wired. the column name should have only one part